website/project/model.py

```diff
@@ -1213,6 +1233,51 @@ def nodes_primary(self):
         ]
 
     @property
+    def depth(self):
+        return len(self.parents)
+
+    def next_descendants(self, auth, condition=lambda auth, node: True):
+        """
+        Recursively find the first set of descedants under a given node that meet a given condition
+
+        returns a list of [(node, [children]), ...]
+        """
+        ret = []
```

Comment:
At the moment, the usage is to find the first set of descendants that fulfill a certain condition, i.e. find the minimal subtree s.t. all leaf nodes fulfill some condition. 

Perhaps the function should be renamed, but I'm not sure how a generator would work here.